src/kaleidoscope/key_defs.h

```diff
@@ -202,6 +203,45 @@ typedef kaleidoscope::Key Key_;
 #define ___ Key_Transparent
 #define XXX Key_NoKey
 
+namespace kaleidoscope {
+constexpr bool isKeyboardKey(Key key) {
+  return ((key.getFlags() & (RESERVED | SYNTHETIC)) == 0);
+}
+constexpr bool isKeyboardModifier(Key key) {
+  return (isKeyboardKey(key) &&
+          (key.getKeyCode() >= HID_KEYBOARD_FIRST_MODIFIER &&
+           key.getKeyCode() <= HID_KEYBOARD_LAST_MODIFIER));
+}
+constexpr bool isKeyboardShift(Key key) {
+  return (isKeyboardModifier(key) &&
```

Comment:
If I'm reading this right, this method will only trigger if the key is:

* a keyboard key
* a modifier key
and 
* ( is a shift key OR is another modifer key with the shift bit set)

That last bit was momentarily surprising, but feels like the correct choice for the question we want to ask.   This is a place where doc explaining what's expected probably makes sense.

"isKeyboardShift" still feels like the wrong name. But I think we should talk the names through not as inline comments.

Reply:
Yes, exactly (I definitely should have written explanatory comments here). Basically, the assumption is that `LCTRL(Key_LeftShift)` and `LSHIFT(Key_LeftControl)` are two different expressions that indicate the same intention on the part of the user: a key that's both `control` and `shift`. So it shouldn't matter which one is in the keycode byte and which one is in the mod flags; for the purpose of this function both are modifier keys that include `shift`.

I've been thinking about the `shift` that's part of a combo modifier like this as an "intentional shift", and one that's a mod flag on a non-modifier (e.g. `LSHIFT(Key_1)`) as an "incidental shift". There are at least a few plugins that should make a distinction between the two, and this is meant to help with that.

Also, it's worth noting that `LSHIFT(Key_NoKey)` will not return `true` from this function, and that's intentional, as well. It will allow, for instance, OneShot with the forthcoming "auto one-shot modifiers" option to not identify that key as an auto-one-shot key.